client/src/components/Sidebar.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+
+const Sidebar = (props) => {
+  const { options, setOptions, labels } = props;
+  const {
+    hideDone, timeRange, filterLabels, ThemeColor,
+  } = options;
+
+  function changeLabelFilter(e) {
+    const { id } = e.target;
+    const labelToChange = filterLabels.find((label) => label.name === id);
+    labelToChange.active = !labelToChange.active;
+    setOptions(options);
+  }
+  function ChangeDisplayClosed() {
+    hideDone.active = !hideDone.active;
+    setOptions(options);
+  }
+  function changeTimeRange(e) {
+    const { checked, value } = e.target;
+    if (e.target.id === 'timeRangeCheckbox') {
+      timeRange.active = checked;
+      setOptions(options);
+    } else if (timeRange.active) {
+      const timeRanges = [
+        'Last 24 Hours',
+        'Last Week',
+        'Last Month',
+        'Last Year',
+        'Last 2 Years',
+        'Last 5 Years',
+        'Always',
+      ];
+      timeRange.range = timeRanges[value];
+      setOptions(options);
+    }
+  }
+  function changeColour(e) {
+    const { target } = e;
+    const value = target.type === 'color'
+      ? target.value
+      : target.style.background;
+    ThemeColor.color = value;
+    setOptions(options);
+  }
+
+  return (
+    <div id="sidebar" className={options.displayMenu ? 'open' : 'closed'}>
+      <div id="themeColorOptions" className="optionContainer">
+        <strong>Pick a Colour preset</strong>
+        <br />
+        <input type="button" value="Sea Green" style={{ background: '#2e8b57', color: 'white' }} onClick={changeColour} />
+        <br />
+        <input type="button" value="Deep Blue" style={{ background: '#313fb8', color: 'white' }} onClick={changeColour} />
+        <br />
+        <input type="button" value="Calming Teal" style={{ background: '#00a4a4', color: 'white' }} onClick={changeColour} />
+        <br />
+        <input type="button" value="Cheerful Pink" style={{ background: '#ffb6c1', color: 'white' }} onClick={changeColour} />
+        <br />
+        <input type="button" value="Chilled Sangria" style={{ background: '#8b0000', color: 'white' }} onClick={changeColour} />
```

Comment:
a lot of duplicate code ! maybe create new component called `<ColorOption value="Cheerful Pink" style={{ background: '#ffb6c1', color: 'white' }} />`